processor/tests/src/test/java/org/mobilitydata/gtfsvalidator/processor/tests/EnumSizesSchemaTest.java

```diff
@@ -0,0 +1,92 @@
+package org.mobilitydata.gtfsvalidator.processor.tests;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mobilitydata.gtfsvalidator.table.ByteGtfs;
+import org.mobilitydata.gtfsvalidator.table.EnumSizes;
+import org.mobilitydata.gtfsvalidator.table.IntGtfs;
+import org.mobilitydata.gtfsvalidator.table.ShortGtfs;
+
+@RunWith(JUnit4.class)
+public class EnumSizesSchemaTest {
+
+  @Test
+  public void testByteEnumValid() {
+    assertThat(new EnumSizes.Builder().setByteEnum(ByteGtfs.ONE).build().byteEnum())
+        .isEqualTo(ByteGtfs.ONE);
+    assertThat(new EnumSizes.Builder().setByteEnum(ByteGtfs.MAX_BYTE).build().byteEnum())
+        .isEqualTo(ByteGtfs.MAX_BYTE);
+    assertThat(new EnumSizes.Builder().setByteEnum(ByteGtfs.MIN_BYTE).build().byteEnum())
+        .isEqualTo(ByteGtfs.MIN_BYTE);
+  }
+
+  @Test
+  public void testByteEnumInvalid() {
+    EnumSizes enumSizes = new EnumSizes.Builder().setByteEnum(10).build();
+    // 10 is not a valid constant for ByteGtfs.
+    assertThat(enumSizes.byteEnum()).isEqualTo(ByteGtfs.UNRECOGNIZED);
+    // However, 10 can be still retrieved using byteEnumValue method.
+    assertThat(enumSizes.byteEnumValue()).isEqualTo(10);
+  }
+
+  @Test
+  public void testByteEnumOutOfBounds() {
+    // Short.MIN_VALUE is out of bounds for byte. It cannot be retrieved with byteEnumValue.
+    assertThat(new EnumSizes.Builder().setByteEnum((int) Short.MIN_VALUE).build().byteEnumValue())
+        .isEqualTo(0);
```

Comment:
Should we throw any exception if values can't be retrieved? It looks like an undefined behaviour

Reply:
We generate UnexpectedEnumValueNotice and store UNRECOGNIZED.